prowler/providers/aws/lib/security_hub/security_hub.py

```diff
@@ -284,3 +285,69 @@ def _send_findings_in_batches(
                 f"{error.__class__.__name__} -- [{error.__traceback__.tb_lineno}]:{error} in region {region}"
             )
             return success_count
+
+    @staticmethod
+    def test_connection(
+        session: Session,
+        region: str,
+        aws_account_id: str,
+        aws_partition: str,
+        raise_on_exception: bool = True,
+    ) -> Connection:
+        """
+        Test the connection to AWS Security Hub by checking if Security Hub is enabled in the provided region
+        and if the Prowler integration is active.
+
+        Args:
+            session (Session): AWS session to use for authentication.
+            region (str): The AWS region to test Security Hub connection.
+            aws_account_id (str): AWS account ID to check for Prowler integration.
+            aws_partition (str): AWS partition (e.g., aws, aws-cn, aws-us-gov).
+            raise_on_exception (bool): Whether to raise an exception if an error occurs.
+
+        Returns:
+            Connection: An object that contains the result of the test connection operation.
+                - is_connected (bool): Indicates whether the connection was successful.
+                - error (Exception): An exception object if an error occurs during the connection test.
+        """
+        try:
+            client = session.client("securityhub", region_name=region)
+
+            # Check if Security Hub is enabled in the region
+            client.describe_hub()
+
+            # Check if Prowler integration is enabled
+            prowler_integration_arn = f"arn:{aws_partition}:securityhub:{region}:{aws_account_id}:product-subscription/{SECURITY_HUB_INTEGRATION_NAME}"
+            enabled_products = client.list_enabled_products_for_import()
+
+            if prowler_integration_arn not in str(enabled_products):
+                logger.warning(
+                    f"Prowler integration is not enabled in region {region}."
+                )
+                return Connection(is_connected=False, error=None)
+
+            return Connection(is_connected=True)
+
+        except ClientError as client_error:
+            error_code = client_error.response["Error"]["Code"]
+            if error_code == "InvalidAccessException":
```

Comment:
Can you validate it getting the `boto3` exception class? Then create a custom exception with the original within.